test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -403,17 +410,113 @@ TEST_F(AuthenticatorTest, TestNoForwardPayloadHeader) {
 
   MockUpstream mock_pubkey(mock_factory_ctx_.cluster_manager_, PublicKey);
   auto headers = Http::TestHeaderMapImpl{{"Authorization", "Bearer " + std::string(GoodToken)}};
-  MockAuthenticatorCallbacks mock_cb;
-  EXPECT_CALL(mock_cb, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+  std::function<void(const Status&)> mock_cb = [](const Status& status) {
     ASSERT_EQ(status, Status::Ok);
-  }));
-  auth_->verify(headers, &mock_cb);
+  };
+  auto tokens = filter_config_->getExtractor().extract(headers);
+  auth_->verify(headers, std::move(tokens), std::move(mock_cb));
 
   // Test when forward_payload_header is not set, the output should NOT
   // contain the sec-istio-auth-userinfo header.
   EXPECT_FALSE(headers.has("sec-istio-auth-userinfo"));
 }
 
+// This test verifies that path match prefix, and JWT authenticates.
+TEST_F(AuthenticatorTest, TestPrefixPathMatch) {
+  auto& rule = *proto_config_.add_rules();
+  rule.mutable_match()->set_prefix("/abc");
+  rule.mutable_requires()->set_provider_name(ProviderName);
+  CreateAuthenticator();
+
+  MockUpstream mock_pubkey(mock_factory_ctx_.cluster_manager_, PublicKey);
+
+  auto headers = Http::TestHeaderMapImpl{
+      {"Authorization", "Bearer " + std::string(GoodToken)},
+      {":path", "/abc/xyz"},
+  };
+
+  std::function<void(const Status&)> mock_cb = [](const Status& status) {
+    ASSERT_EQ(status, Status::Ok);
+  };
+  auto tokens = filter_config_->getExtractor().extract(headers);
+  auth_->verify(headers, std::move(tokens), std::move(mock_cb));
+
+  EXPECT_EQ(headers.get_("sec-istio-auth-userinfo"), ExpectedPayloadValue);
+  // Verify the token is removed.
+  EXPECT_FALSE(headers.Authorization());
+  EXPECT_EQ(mock_pubkey.called_count(), 1);
+}
+
+// This test verifies that path match regex, and JWT authenticates.
+TEST_F(AuthenticatorTest, TestRegexPathMatch) {
+  auto& rule = *proto_config_.add_rules();
+  rule.mutable_match()->set_regex("/b[oi]t");
+  rule.mutable_requires()->set_provider_name(ProviderName);
```

Comment:
same comment as last test

Reply:
These are old tests that are no longer needed. I will remove them.